Source: Fedora/Upstream http://pkgs.fedoraproject.org/cgit/cups-filters.git/commit/?id=9261356b85fcd4861e432b70d34abaf126c22b9c
Reason: Fix compiling against poppler 0.22.x
Upstream: Merged

diff -Naur cups-filters-1.0.29.orig/filter/pdftoopvp/oprs/OPRS.cxx cups-filters-1.0.29/filter/pdftoopvp/oprs/OPRS.cxx
--- cups-filters-1.0.29.orig/filter/pdftoopvp/oprs/OPRS.cxx	2012-09-10 21:52:00.000000000 +0200
+++ cups-filters-1.0.29/filter/pdftoopvp/oprs/OPRS.cxx	2013-03-05 00:53:05.532490631 +0100
@@ -5,6 +5,9 @@
 //========================================================================
 
 #include <config.h>
+#ifdef HAVE_CPP_POPPLER_VERSION_H
+#include "cpp/poppler-version.h"
+#endif
 
 #ifdef USE_GCC_PRAGMAS
 #pragma implementation
@@ -235,7 +238,11 @@
 			      SplashColorMode srcMode, GBool srcAlpha,
 			      int w, int h, SplashCoord *mat) {
     if (rasterMode) {
+#if POPPLER_VERSION_MAJOR <= 0 && (POPPLER_VERSION_MINOR <= 20 || (POPPLER_VERSION_MINOR == 21 && POPPLER_VERSION_MICRO <= 2))
 	return splash->drawImage(src,srcData,srcMode,srcAlpha,w,h,mat);
+#else
+	return splash->drawImage(src,srcData,srcMode,srcAlpha,w,h,mat,gFalse);
+#endif
     } else {
 	return opvpSplash->drawImage(src,srcData,srcMode,srcAlpha,w,h,mat);
     }
diff -Naur cups-filters-1.0.29.orig/filter/pdftoopvp/OPVPOutputDev.cxx cups-filters-1.0.29/filter/pdftoopvp/OPVPOutputDev.cxx
--- cups-filters-1.0.29.orig/filter/pdftoopvp/OPVPOutputDev.cxx	2012-07-20 10:22:23.000000000 +0200
+++ cups-filters-1.0.29/filter/pdftoopvp/OPVPOutputDev.cxx	2013-03-05 00:55:27.154477349 +0100
@@ -1804,8 +1804,13 @@
   maskSplash = new Splash(maskBitmap, gFalse);
   maskColor[0] = 0;
   maskSplash->clear(maskColor);
+#if POPPLER_VERSION_MAJOR <= 0 && (POPPLER_VERSION_MINOR <= 20 || (POPPLER_VERSION_MINOR == 21 && POPPLER_VERSION_MICRO <= 2))
   maskSplash->drawImage(&imageSrc, &imgMaskData,
 			splashModeMono8, gFalse, maskWidth, maskHeight, mat);
+#else
+  maskSplash->drawImage(&imageSrc, &imgMaskData,
+			splashModeMono8, gFalse, maskWidth, maskHeight, mat, gFalse);
+#endif
   delete imgMaskData.imgStr;
   maskStr->close();
   gfree(imgMaskData.lookup);
